### DISCUSSION TITLE
Where can I find the 'fine-tuning' branch?

### DISCUSSION THREAD

**Discussion by turbosonics** [ACEsuit/mace#471] [Q&A] [answered]
From one of past discussions: 
https://github.com/ACEsuit/mace/discussions/254

I also hope to test fine-tuning, but I also faced the same problem: The current MACE I compiled two months ago doesn't work with "--foundation_model". But I couldn't find any further information about 'fine-tuning' branch of MACE. 

If normal branch and 'fine-tuning' branch are not merged yet, then where can I find the 'fine-tuning' branch? I assume the installation would be the same, but I couldn't find any link or direction for 'fine-tuning' branch. Any assistance would be appreciated. 

Thanks

**Comment by ilyes319** [comment, marked as answer]
You should be able to do normal finetuning on the main branch now. What error do you have?

**Reply by turbosonics** [reply]
Thanks for your reply. I tried with the old version that I compiled months ago. Let me recompile the current version and check. Thanks.

**Reply by turbosonics** [reply]
I confirm that the fine tuning with '--foundation_model' worked with MACE version of June 21 2024. Thanks for your help.

**Reply by nitbha007** [reply]
Hello @ilyes319 , I was fine-tuning one of my own models by assigning the path of it to the "--foundation_model" argument. I encountered the following error though:
```
[rank0]: Traceback (most recent call last):
[rank0]:   File "/projappl/Nitik_work/Normal_mace/run_train.py", line 850, in <module>
[rank0]:     main()
[rank0]:   File "/projappl/Nitik_work/Normal_mace/run_train.py", line 51, in main
[rank0]:     run(args)
[rank0]:   File "/projappl/Nitik_work/Normal_mace/run_train.py", line 513, in run
[rank0]:     model = load_foundations(
[rank0]:   File "/PUHTI_TYKKY_20dcsj1/miniconda/envs/env1/lib/python3.10/site-packages/mace/tools/finetuning_utils.py", line 120, in load_foundations
[rank0]:     model_foundations.products[i]
[rank0]:   File "/PUHTI_TYKKY_20dcsj1/miniconda/envs/env1/lib/python3.10/site-packages/torch/nn/modules/container.py", line 606, in __getitem__
[rank0]:     idx = self._get_abs_string_index(idx)
[rank0]:   File "/PUHTI_TYKKY_20dcsj1/miniconda/envs/env1/lib/python3.10/site-packages/torch/nn/modules/container.py", line 585, in _get_abs_string_index
[rank0]:     raise IndexError(f'index {idx} is out of range')
[rank0]: IndexError: index 1 is out of range

```
I provided the same hyper-parameters that I used in training the initial MACE model. I have also tested with the foundation models like "small", fine-tuning was happening fine but not with my own model. 

Thank you!

**Reply by ilyes319** [reply]
What are the hypers of your own model? If you are using correlation order 2, it will not work. In any case, we discourage people to use correlation order 2.

**Reply by nitbha007** [reply]
Well, I did use the correlation order 2! Any possibility to make it work around it or other solutions?

**Reply by ilyes319** [reply]
What lead you tu use correlation order 2?

**Reply by nitbha007** [reply]
I did not find much difference in the accuracy on increasing the correlation order. Perhaps I can train the model again with higher correlation order and use that for fine-tuning.
